src/Symfony/Component/Serializer/Exception/AggregatedException.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Exception;
+
+final class AggregatedException extends RuntimeException implements AggregableExceptionInterface
+{
+    /**
+     * @var bool
+     */
+    public $has = false;
+
+    /**
+     * @var array
+     */
+    private $collection;
+
+    public function __construct(\Throwable $previous = null)
+    {
+        parent::__construct('Errors occurred during the denormalization process', 0, $previous);
+    }
+
+    public function add(string $param, AggregableExceptionInterface $exception): self
+    {
+        $this->has = true;
+        $this->collection[] = [$param, $exception];
+
+        return $this;
+    }
+
+    /**
+     * @return \Generator<string, AggregableExceptionInterface>
+     */
+    public function getExceptions(): \Generator
```

Comment:
I guess you choose to use a generator so you could have multiple keys with the same value.
And you said it was for the variadic parameters, what about making each key unique: `paramName.<index>` ?
This would allow to have an output like:
```
[
    'variadicParam.2' => 'Some error',
    'variadicParam.5' => 'Another error',
]
```

Because if for some reason the error message does not include information about the data it might not be possible to determine which of the value is invalid.

Reply:
Fixed. Use `array<string, AggregableExceptionInterface>` instead of `Generator`.